### PR TITLE
Update connect cycle 

### PR DESCRIPTION
**Pull request by kaylareopelle**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
* Turn on distributed tracing by default 
* Increase `span_events.max_samples_stored` default value from 1000 to 2000
* Update connect cycle to incorporate latest agent specs span event harvest data definitions

# Related Github Issue
Closes: https://github.com/newrelic/newrelic-ruby-agent/issues/755 
Closes: https://github.com/newrelic/newrelic-ruby-agent/issues/550

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
Github Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-18.04
     strategy:
       matrix:
-        ruby-version: [2.2.10, 2.3.8, 2.4.10, 2.5.9, 2.6.8, 2.7.4, 3.0.2, jruby-9.2.17.0]
+        ruby-version: [2.2.10, 2.3.8, 2.4.10, 2.5.9, 2.6.8, 2.7.4, 3.0.2, jruby-9.2.19.0]
```
Comment (kaylareopelle) [inline .github/workflows/ci.yml:17]: @amhuntsman - Though this change is also in httprb PR #722, we wanted to add it here to see if it helped with some odd test failures that appear on jruby's rack multiverse tests, but nowhere else.

Comment (kaylareopelle) [reply .github/workflows/ci.yml:17]: It didn't hurt those failures, so we'd like to keep this change in the PR if it doesn't create too much of a conflict.

File: lib/new_relic/agent/configuration/event_harvest_config.rb [hunk 2]
```diff
@@ -27,16 +28,31 @@ def from_config(config)
...
+            event_harvest_config
+          end
+        end
+
+        def transform_span_event_harvest_config(config_hash, connect_reply)
+          if span_harvest = connect_reply['span_event_harvest_config']
+            config_hash[:'span_events.max_samples_stored'] = span_harvest['harvest_limit'] if span_harvest['harvest_limit']
+            config_hash[:'event_report_period.span_event_data'] = span_harvest['report_period_ms'] if span_harvest['report_period_ms']
+          end
+
```
Comment (amhuntsman) [inline lib/new_relic/agent/configuration/event_harvest_config.rb:55]: Is there a cliffnotes version of these changes? This file is pretty hard to read as is.

File: test/new_relic/marshalling_test_cases.rb [hunk 3]
```diff
@@ -137,15 +134,20 @@ def test_sends_error_events
...
+    assert_equal t0.to_f, event[0]["timestamp"]
+    assert_equal "TestTransaction/break_it", event[0]["transactionName"]
+    assert_equal 0.0, event[0]["duration"]
+    assert_equal "Unknown", event[0]["parent.transportType"]
     assert event[0]["spanId"] != nil
-    assert_equal event[0].size, 8
+    assert event[0]['guid'] != nil
+    assert event[0]["traceId"] != nil
+
+    assert_equal 12, event[0].size
```
Comment (amhuntsman) [inline test/new_relic/marshalling_test_cases.rb:149]: 🎉

**Review by amhuntsman — APPROVED** [review]

File: lib/new_relic/agent/configuration/event_harvest_config.rb (hunk 2 above)
Comment (kaylareopelle) [reply lib/new_relic/agent/configuration/event_harvest_config.rb:55]: Sure thing! 🧗 

**Big change:** We needed to accommodate for the new way span event data is structured according to the  [agent spec changes to the span event harvest config](https://source.datanerd.us/agents/agent-specs/blob/master/Connect-LEGACY.md#span-event-harvest-config).

We decided to accomplish that by pulling it out of the constant and adding it back into the event harvest config using its new format in `#from_config` and `#to_config_hash`. The `#to_config_hash` method was getting a little long, so we refactored it into two methods to format the hash.

**Pull request merged**